connect/runtime/src/main/java/org/apache/kafka/connect/storage/ConnectorOffsetBackingStore.java

```diff
@@ -302,7 +326,12 @@ public Future<Void> set(Map<ByteBuffer, ByteBuffer> values, Callback<Void> callb
                 }
             }
             try (LoggingContext context = loggingContext()) {
-                callback.onCompletion(primaryWriteError, ignored);
+                Throwable secondaryWriteError = secondaryStoreTombstoneWriteError.get();
+                if (secondaryStore != null && containsTombstones && secondaryWriteError != null) {
```

Comment:
Same as above - we aren't blocking on the write to the secondary store, so we can't be sure that it has completed at this point.

Reply:
That's true and I have tweaked the logic to wait for secondary store write in case of tombstone records before proceeding with primary store writes.